html/mycakeapp/src/Controller/AuctionRatingController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Controller;
+
+use App\Controller\AppController;
+
+use Cake\Event\Event;
+use Exception;
+
+class AuctionRatingController extends AuctionBaseController
+{
+
+    public $useTable = false;
+
+    public function initialize()
+    {
+        parent::initialize();
+        $this->loadComponent('Paginator');
+        $this->loadModel('Users');
+        $this->loadModel('Biditems');
+        $this->loadModel('Bidinfo');
+        $this->loadModel('Rates');
+        $this->set('authuser', $this->Auth->user());
+        $this->viewBuilder()->setLayout('auction');
+    }
+
+    public function addRate($bidinfo_id = null)
+    {
+        try {
+            $bidinfo = $this->Bidinfo->get($bidinfo_id, [
+                'contain' => ['Biditems']
+            ]);
+            $seller = $bidinfo->biditem->user_id;
+            $buyer = $bidinfo->user_id;
+            // この取引を評価済みであればインデックスへリダイレクトする
+            $is_rated = $this->Rates->find('all', [
+                'conditions' => ['rater_id' => $this->Auth->user('id'), 'bidinfo_id' => $bidinfo_id]
+            ])->first();
+            if ($is_rated ?? false) {
```

Comment:
(修正不要)
```suggestion
            if ($is_rated) {
```
変な癖ついてますね 笑